adder/single/dag_service.go

```diff
@@ -30,22 +30,28 @@ type DAGService struct {
 
 	dests   []peer.ID
 	pinOpts api.PinOptions
+	local   bool
 
 	ba *adder.BlockAdder
 }
 
 // New returns a new Adder with the given rpc Client. The client is used
 // to perform calls to IPFS.BlockPut and Pin content on Cluster.
-func New(rpc *rpc.Client, opts api.PinOptions) *DAGService {
+func New(rpc *rpc.Client, opts api.PinOptions, local bool) *DAGService {
 	return &DAGService{
 		rpcClient: rpc,
 		dests:     nil,
 		pinOpts:   opts,
+		local:     local,
 	}
 }
 
 // Add puts the given node in the destination peers.
 func (dgs *DAGService) Add(ctx context.Context, node ipld.Node) error {
+	if dgs.local {
+		return adder.AddNodeLocal(ctx, dgs.rpcClient, node)
+	}
+
```

Comment:
you need to do the blockallocate even if you don't use it to send the blocks to those destinations because you are running on local node.

Reply:
Below, if you are in local-mode, your BlockAdder destinations are []("") (or better, modify the block adder to have the local parameter). There is no reason to add new code to add blocks to a place when you already have code to do that.

Reply:
can do this by passing an empty dest